### PR TITLE
Store Secrets to Encrypted Storage

### PR DESCRIPTION
**Pull request by msJinLei**
## Description
- Implement AzKeyStore to support multiple data structures of keys and values
- Store ServicePrincipalSecret and CertificatePassword into AzKeyStore
- Add test cases

## Checklist

- [ ] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/Accounts/Accounts.Test/AutosaveTests.cs [hunk 1]
```diff
@@ -22,23 +22,38 @@
...
     {
         private MemoryDataStore dataStore;
         private MockCommandRuntime commandRuntimeMock;
+        private AzKeyStore keyStore;
         public AutosaveTests(ITestOutputHelper output)
         {
             XunitTracingInterceptor.AddToContext(new XunitTracingInterceptor(output));
             commandRuntimeMock = new MockCommandRuntime();
             dataStore = new MemoryDataStore();
-            ResetState();
```
Comment (BethanyZhou) [inline src/Accounts/Accounts.Test/AutosaveTests.cs:41]: ResetState() never invokes, is it expected

File: src/Accounts/Authentication/KeyStore/AzKeyStore.cs [hunk 2]
```diff
@@ -11,67 +11,220 @@
...
+            Storage = inputStorage ?? new StorageWrapper()
+            {
+                FileName = fileName,
+                Directory = directory
+            };
+            Storage.Create();
+
+            if (loadStorage)
+            {
+                if (!LoadStorage())
```
Comment (BethanyZhou) [inline src/Accounts/Authentication/KeyStore/AzKeyStore.cs:91]: is this an async operation?

File: src/Accounts/Accounts.Test/AutosaveTests.cs (hunk 1 above)
Comment (msJinLei) [reply src/Accounts/Accounts.Test/AutosaveTests.cs:41]: It is called at the beginning and end of the every test cases

File: src/Accounts/Authentication/KeyStore/AzKeyStore.cs (hunk 2 above)
Comment (msJinLei) [reply src/Accounts/Authentication/KeyStore/AzKeyStore.cs:91]: It is a lightweight storage. I think we can sync mode here.

File: src/Accounts/Authentication/KeyStore/AzKeyStore.cs [hunk 3]
```diff
@@ -11,67 +11,220 @@
...
+                {
+                    throw new InvalidOperationException("Failed to load keystore from storage.");
+                }
+            }
```
Comment (BethanyZhou) [inline src/Accounts/Authentication/KeyStore/AzKeyStore.cs:95]:
```suggestion
            if (loadStorage&&!LoadStorage())
            {
                throw new InvalidOperationException("Failed to load keystore from storage.");
            }
```

**Review by BethanyZhou — APPROVED** [review]

**Pull request merged**